libcontainer/utils/cmsg.go

```diff
@@ -61,36 +57,63 @@ func RecvFd(socket *os.File) (*os.File, error) {
 	if err != nil {
 		return nil, err
 	}
-	if len(scms) != 1 {
-		return nil, fmt.Errorf("recvfd: number of SCMs is not 1: %d", len(scms))
+
+	// We cannot control how many SCM_RIGHTS we receive, and upon receiving
+	// them all of the descriptors are installed in our fd table, so we need to
+	// parse all of the SCM_RIGHTS we received in order to close all of the
+	// descriptors on error.
+	var fds []int
+	defer func() {
+		for i, fd := range fds {
+			if i == 0 && err == nil {
+				// Only close the first one on error.
+				continue
+			}
+			// Always close extra ones.
+			_ = unix.Close(fd)
```

Comment:
@cyphar I slightly changed the code here; the original was this:
```go
			// Only close 0 if err != nil, and close everything else.
			if i != 0 || err != nil {
				_ = unix.Close(fd)
			}
```

I feel that my version is less compact but more readable. Feel free to 👎🏻 and I will revert :)

Reply:
Ah! I wanted to fix the error but forgot (the error is we should check `Err` not `err`). Fixed now.